consul/consul.go

```diff
@@ -203,26 +205,49 @@ func (c *Consul) marathonTaskToConsulService(task *apps.Task, app *apps.App) (*c
 		Port:    task.Ports[0],
 		Address: serviceAddress,
 		Tags:    c.marathonLabelsToConsulTags(app.Labels),
-		Check:   c.marathonToConsulCheck(task, app.HealthChecks, serviceAddress),
+		Checks:  c.marathonToConsulChecks(task, app.HealthChecks, serviceAddress),
 	}, nil
 }
 
-func (c *Consul) marathonToConsulCheck(task *apps.Task, healthChecks []apps.HealthCheck, serviceAddress string) *consulapi.AgentServiceCheck {
-	//	TODO: Handle all types of checks
+func (c *Consul) marathonToConsulChecks(task *apps.Task, healthChecks []apps.HealthCheck, serviceAddress string) consulapi.AgentServiceChecks {
+
+	var checks consulapi.AgentServiceChecks = make(consulapi.AgentServiceChecks, 0)
+
 	for _, check := range healthChecks {
-		if check.Protocol == "HTTP" {
-			return &consulapi.AgentServiceCheck{
+		switch check.Protocol {
+		case "HTTP", "HTTPS":
+			checks = append(checks, &consulapi.AgentServiceCheck{
 				HTTP: (&url.URL{
-					Scheme: "http",
-					Host:   fmt.Sprintf("%s:%d", serviceAddress, task.Ports[check.PortIndex]),
-					Path:   check.Path,
+					Scheme:  strings.ToLower(check.Protocol),
+					Host:    fmt.Sprintf("%s:%d", serviceAddress, task.Ports[check.PortIndex]),
+					RawPath: check.Path,
 				}).String(),
 				Interval: fmt.Sprintf("%ds", check.IntervalSeconds),
 				Timeout:  fmt.Sprintf("%ds", check.TimeoutSeconds),
-			}
+			})
+		case "TCP":
+			checks = append(checks, &consulapi.AgentServiceCheck{
+				TCP:      fmt.Sprintf("%s:%d", serviceAddress, task.Ports[check.PortIndex]),
+				Interval: fmt.Sprintf("%ds", check.IntervalSeconds),
+				Timeout:  fmt.Sprintf("%ds", check.TimeoutSeconds),
+			})
+
+		case "COMMAND":
+			checks = append(checks, &consulapi.AgentServiceCheck{
+				Script:   check.Command.Value,
+				Interval: fmt.Sprintf("%ds", check.IntervalSeconds),
+				Timeout:  fmt.Sprintf("%ds", check.TimeoutSeconds),
+			})
+		default:
+			log.WithField("Id", task.AppID.String()).WithField("Address", serviceAddress).
+				Error(fmt.Printf("Unrecognized check protocol %s", check.Protocol))
 		}
+
 	}
-	return nil
+
+	spew.Dump(checks)
```

Comment:
Hold on :) I'm trying to test this PR as on my macbook tests are always failing :palm_tree: